src/main/kotlin/rx/lang/kotlin/observables.kt

```diff
@@ -2,71 +2,60 @@ package rx.lang.kotlin
-    override fun iterator(): Iterator<T> = this@toIterable
-}
-
-fun BooleanArray.toObservable() : Observable<Boolean> = this.toList().toObservable()
-fun ByteArray.toObservable() : Observable<Byte> = this.toList().toObservable()
-fun ShortArray.toObservable() : Observable<Short> = this.toList().toObservable()
-fun IntArray.toObservable() : Observable<Int> = this.toList().toObservable()
-fun LongArray.toObservable() : Observable<Long> = this.toList().toObservable()
-fun FloatArray.toObservable() : Observable<Float> = this.toList().toObservable()
-fun DoubleArray.toObservable() : Observable<Double> = this.toList().toObservable()
-fun <T> Array<out T>.toObservable() : Observable<T> = Observable.from(this)
-
-fun IntProgression.toObservable() : Observable<Int> =
+fun BooleanArray.toObservable(): Observable<Boolean> = this.toList().toObservable()
+fun ByteArray.toObservable(): Observable<Byte> = this.toList().toObservable()
+fun ShortArray.toObservable(): Observable<Short> = this.toList().toObservable()
+fun IntArray.toObservable(): Observable<Int> = this.toList().toObservable()
+fun LongArray.toObservable(): Observable<Long> = this.toList().toObservable()
+fun FloatArray.toObservable(): Observable<Float> = this.toList().toObservable()
+fun DoubleArray.toObservable(): Observable<Double> = this.toList().toObservable()
+fun <T> Array<out T>.toObservable(): Observable<T> = Observable.from(this)
+
+fun IntProgression.toObservable(): Observable<Int> =
         if (step == 1 && last.toLong() - first < Integer.MAX_VALUE) Observable.range(first, Math.max(0, last - first + 1))
         else Observable.from(this)
 
-fun <T> Iterator<T>.toObservable() : Observable<T> = toIterable().toObservable()
-fun <T> Iterable<T>.toObservable() : Observable<T> = Observable.from(this)
-fun <T> Sequence<T>.toObservable() : Observable<T> = Observable.from(object : Iterable<T> {
+fun <T> Iterator<T>.toObservable(): Observable<T> = toIterable().toObservable()
+fun <T> Iterable<T>.toObservable(): Observable<T> = Observable.from(this)
+fun <T> Sequence<T>.toObservable(): Observable<T> = Observable.from(object : Iterable<T> {
     override fun iterator(): Iterator<T> = this@toObservable.iterator()
 })
 
-fun <T> T.toSingletonObservable() : Observable<T> = Observable.just(this)
-fun <T> Throwable.toObservable() : Observable<T> = Observable.error(this)
+fun <T> Throwable.toObservable(): Observable<T> = Observable.error(this)
 
-fun <T> Iterable<Observable<out T>>.merge() : Observable<T> = Observable.merge(this.toObservable())
-fun <T> Iterable<Observable<out T>>.mergeDelayError() : Observable<T> = Observable.mergeDelayError(this.toObservable())
+fun <T> Iterable<Observable<out T>>.merge(): Observable<T> = Observable.merge(this.toObservable())
+fun <T> Iterable<Observable<out T>>.mergeDelayError(): Observable<T> = Observable.mergeDelayError(this.toObservable())
 
+fun <T, R> Observable<T>.fold(initial: R, body: (R, T) -> R): Observable<R> = reduce(initial, { a, e -> body(a, e) })
```

Comment:
We'll revisit that idea if it makes sense later, or if there is significant demand for it despite our cautions.